Content.Server/Discord/DiscordLink.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System.Diagnostics;
+using System.Threading.Tasks;
+using Content.Shared.CCVar;
+using Discord;
+using Discord.WebSocket;
+using Robust.Shared.Configuration;
+using LogMessage = Discord.LogMessage;
+
+namespace Content.Server.Discord;
+
+public sealed class DiscordLink : IPostInjectInit
+{
+    [Dependency] private readonly ILogManager _logManager = default!;
+    [Dependency] private readonly IConfigurationManager _configuration = default!;
+
+    public DiscordSocketClient? Client;
+    private ISawmill _sawmill = default!;
+
+    private string _guildId = string.Empty;
+
+    public void PostInject()
+    {
+        _sawmill = _logManager.GetSawmill("discord.link");
+
+        Client = new DiscordSocketClient(new DiscordSocketConfig()
+        {
+            GatewayIntents = GatewayIntents.All // I FUCKIGN hate this. I do not care enough for me to actually set proper intents. TODO: Change this I guess?
+        });
+        Client.Log += Log;
+
+        try
+        {
+            _configuration.OnValueChanged(CCVars.DiscordGuildId, OnGuildIdChanged, true);
+        }
+        catch (Exception e)
+        {
+            _sawmill.Fatal($"Failed to setup cvar.");
+            return;
+        }
+
+        if (_configuration.GetCVar(CCVars.DiscordToken) is not { } token)
+        {
+            _sawmill.Warning("No Discord token specified, not connecting.");
```

Comment:
This should not be a warning case, it is a totally valid use for game servers to not have the Discord integration configured.